### PR TITLE
Switch Custom Metrics Shipment

### PR DESCRIPTION
**Pull request by KarstenSchnitter**
SAP BTP Application Logging has been changed to accept custom metrics within
the log message stream. For this the messages need to be in JSON format and
contain a field "type":"metrics" and an array of metrics in field "metrics". The prior
approach using an Http endpoint is deprecated for removal. This commit changes
cf-java-logging-support to provide custom metrics according to the new format by
writing the correct JSON to `System.out`.

### CODE REVIEW

File: cf-java-monitoring-custom-metrics-clients/cf-custom-metrics-clients-java/src/main/java/com/sap/cloud/cf/monitoring/java/CustomMetricRegistry.java [hunk 1]
```diff
@@ -37,26 +34,16 @@ protected CustomMetricsReporter getReporter() {
...
-    }
-
     private void initializeAndStartReporter() {
-        if (configProvider == null || customMetricsConfig == null || !customMetricsConfig.isEnabled()) {
+		if (customMetricsConfig == null || !customMetricsConfig.isEnabled()) {
             LOGGER.error("Custom Metrics reporter will not start since required ENVs are missing or environment variable 'enable' is false.");
             return;
         }
-        MonitoringClient client = new MonitoringClientBuilder().setConfigurationProvider(configProvider).create();
+		MonitoringClient client = new MonitoringClientBuilder().create();
```
Comment (nicklas-dohrn) [inline cf-java-monitoring-custom-metrics-clients/cf-custom-metrics-clients-java/src/main/java/com/sap/cloud/cf/monitoring/java/CustomMetricRegistry.java:46]: also formatting error

File: cf-java-monitoring-custom-metrics-clients/cf-custom-metrics-clients-spring-boot/src/test/java/com/sap/cloud/cf/monitoring/spring/configuration/CustomMetricsConditionTest.java [hunk 2]
```diff
@@ -1,97 +1,57 @@
...
+	}
 
     @Test
     public void testMatches_WithAllEnvs() throws Exception {
-        EnvUtils.setEnvs(new String[][] { CF_INSTANCE_GUID_KEY_ENV, CF_INSTANCE_INDEX_KEY_ENV, VCAP_SERVICES_KEY_ENV,
-                VCAP_APPLICATION_KEY_ENV });
+		when(environment.getProperty("VCAP_SERVICES")).thenReturn("{\"application-logs\": [{}]}");
 
-        boolean matches = condition.matches(null, null);
+		boolean matches = condition.matches(context, null);
```
Comment (nicklas-dohrn) [inline cf-java-monitoring-custom-metrics-clients/cf-custom-metrics-clients-spring-boot/src/test/java/com/sap/cloud/cf/monitoring/spring/configuration/CustomMetricsConditionTest.java:35]: format

**Review by nicklas-dohrn — APPROVED** [review]
As far as I can see, the changes seem to be correct.
some small issues with formatting and one maybe debug is still to be corrected.

File: cf-java-monitoring-custom-metrics-clients/cf-custom-metrics-clients-core/src/main/java/com/sap/cloud/cf/monitoring/client/MonitoringClientBuilder.java [hunk 3]
```diff
@@ -92,77 +34,9 @@ public void send(List<Metric> metrics) throws MonitoringClientException {
...
-
-        private UsernamePasswordCredentials getCredentials(ConfigurationProvider provider) {
-            if (credentials == null) {
-                credentials =
-                    new UsernamePasswordCredentials(provider.getClientId(), new String(provider.getClientSecret()));
-            }
-            return credentials;
+			MetricEnvelope envelope = new MetricEnvelope(metrics);
+			String message = gson.toJson(envelope);
+			System.out.println(message);
```
[the remark this answers is not in the document]
Comment (KarstenSchnitter) [reply cf-java-monitoring-custom-metrics-clients/cf-custom-metrics-clients-core/src/main/java/com/sap/cloud/cf/monitoring/client/MonitoringClientBuilder.java:39]: No, this is actually how the metrics message is created.

File: cf-java-monitoring-custom-metrics-clients/cf-custom-metrics-clients-java/src/main/java/com/sap/cloud/cf/monitoring/java/CustomMetricRegistry.java [hunk 4]
```diff
@@ -37,26 +34,16 @@ protected CustomMetricsReporter getReporter() {
...
-            return new CFConfigurationProvider();
-        } catch (IllegalArgumentException e) {
-            LOGGER.error("Required ENVs are missing: {}", e);
-            return null;
-        }
-    }
-
     private void initializeAndStartReporter() {
-        if (configProvider == null || customMetricsConfig == null || !customMetricsConfig.isEnabled()) {
+		if (customMetricsConfig == null || !customMetricsConfig.isEnabled()) {
```
[the remark this answers is not in the document]
Comment (KarstenSchnitter) [reply cf-java-monitoring-custom-metrics-clients/cf-custom-metrics-clients-java/src/main/java/com/sap/cloud/cf/monitoring/java/CustomMetricRegistry.java]: Good catch, I adjusted my Eclipse settings and reformatted the changed files.

**Review by nicklas-dohrn — APPROVED** [review]
LGTM

**Pull request merged**